applications/NXstress.nxdl.xml

```diff
@@ -0,0 +1,1115 @@
+            </doc>       
+        </field>
+        <field name="processing_type" type="NX_CHAR" minOccurs="1" maxOccurs="1">
+            <doc>
+                Describes the way strain :math:`\varepsilon` can be calculated from the :ref:`center &lt;/NXstress/ENTRY/peaks/center-field&gt;`
+                peak parameter.
+            </doc>
+            <enumeration>
+                <item value="two-theta">
+                    <doc>
+                        :math:`\varepsilon = \large \frac{sin(\mathrm{\theta}_{0})}{sin(\mathrm{\theta})}-1`
+                    </doc>
+                </item>
+                <item value="energy">
+                    <doc>
+                        :math:`\varepsilon = \large \frac{\mathrm{E}_{0}}{\mathrm{E}}-1`
+                    </doc>
+                </item>
+                <item value="d-spacing">
+                    <doc>
+                        :math:`\varepsilon = \large \frac{\mathrm{d}}{\mathrm{d}_{0}}-1`
+                    </doc>
+                </item>
+                <item value="time-of-flight">
+                    <doc>
+                        :math:`\varepsilon = \large \frac{\mathrm{TOF}}{\mathrm{TOF}_{0}}-1`
+                    </doc>
+                </item>
+                <item value="sin2psi">
+                    <doc>
+                        A description of the :math:`\mathrm{\sin}^{2}\psi` method can be found in the literature. Two examples are:
+                        `Fitzpatrick, M. E. et. al.; Determination of residual stresses by X-ray diffraction; National Physical Laboratory, Teddington, 2005`
+                        and
+                        `EN 15305:2009 standard: Non-destructive testing - Test Method for Residual Stress analysis by X-ray Diffraction`.
+                    </doc>
+                </item>
+            </enumeration>
+        </field>         
+        <field name="measurement_direction" type="NX_CHAR" minOccurs="0" maxOccurs="1">
+            <doc>
+                Describes the specific measurement direction covered by the data in this file.
+            </doc>
+            <enumeration>
+                <item value="radial"/>
+                <item value="longitudinal"/>
+                <item value="normal"/>
+                <item value="tangential"/>
+                <item value="multiple"/>
+            </enumeration>
+        </field>
+        <group name="experiment_responsible" type="NXuser" minOccurs="0" maxOccurs="1">
+            <doc>Information about the person who performed the experiment.</doc>
+            <field name="name" type="NX_CHAR" minOccurs="0" maxOccurs="1">
+            </field>
+            <field name="role" type="NX_CHAR" minOccurs="0" maxOccurs="1">
+                <doc>Role of user responsible for this entry. Suggested roes are, for example, ``local contact``, ``beamline_scientist``, ``post_doc``,…</doc>

```

Comment:
Is the repetition of docs here and below necessary any more? They become a maintenance burden

Reply:
NXstress has indeed lots of repetition from base classes. The consortium wanted to highlight the fields they deemed most relevant for this application definition. If we can find a way of "highlighting fields/groups to be shown in the application definition" without copying them, that would be great!

Reply:
I think you could just omit the doc - this makes the node optional/requirement/recommend but shown in the application definition. There may be a change by FAIRmat to bubble up the base class definitions' docs in the rendered html/pdf...